components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -143,9 +148,26 @@ public String retrieveArtifact(String APIId, String gatewayLabel, String gateway
                 throw new ArtifactSynchronizerException("HTTP response is empty");
             }
             Map <String, String> apiAttribute = new HashMap<>();
+
             JSONObject artifactObject = new JSONObject(responseString);
-            String apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
-            String label = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABEL);
+            String label = null;
+            String apiId = null;
+            try {
+                apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
+                String labelsStr = artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABELS).toString();
+
+                Set<String> labelsSet = new Gson().fromJson(labelsStr, new TypeToken<HashSet<String>>(){}.getType());
+                Set<String> gatewaySubscribedLabel = gatewayArtifactSynchronizerProperties.getGatewayLabels();
+                if (!labelsSet.isEmpty() || !gatewaySubscribedLabel.isEmpty()){
+                    labelsSet.retainAll(gatewaySubscribedLabel);
+                    if (!labelsSet.isEmpty()){
+                        label = labelsSet.iterator().next();
+                    }
+                }
+            } catch (Exception e ){
```

Comment:
can't we catch specific exception

Reply:
Fixed f1fefa5ebc85177015c7bdac17899ebac013acdd